### PR TITLE
Smooth rough ApiBehavior edges

### LINKED ISSUE #7262: Smooth some rough edges in KodKod implementation
See discussion here - https://github.com/aspnet/Mvc/issues/6828#issuecomment-359091079

We did not get enough time to complete the polish for the new ApiController + Swagger pieces. For preview1, we;ll dial some of these features down until we can get back to fixing them:

1. Make the 'error filter' use the old style of bad request
2. Get rid of the automatic 'default' response we add to api explorer.

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #7262

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.ApiExplorer/ApiBehaviorApiDescriptionProvider.cs [hunk 1]
```diff
@@ -38,20 +38,20 @@ public void OnProvidersExecuting(ApiDescriptionProviderContext context)
                     continue;
                 }
 
-                foreach (var responseType in CreateProblemResponseTypes(description))
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.ApiExplorer/ApiBehaviorApiDescriptionProvider.cs]: I'm not sure we should have this class at all in preview1. This is going to create responses without any type information. Does this override what a user would have configured?

Comment (pranavkm) [reply src/Microsoft.AspNetCore.Mvc.ApiExplorer/ApiBehaviorApiDescriptionProvider.cs]: Discusses offline and we removed this type for preview1.

File: src/Microsoft.AspNetCore.Mvc.Core/Internal/ApiBehaviorOptionsSetup.cs [hunk 2]
```diff
@@ -29,16 +30,16 @@ IActionResult GetInvalidModelStateResponse(ActionContext context)
...
-                    },
-                };
+                    context.ModelState);
+
+                var result = (errorDetails is ModelStateDictionary modelState) ?
+                    new BadRequestObjectResult(modelState) :
+                    new BadRequestObjectResult(errorDetails);
+
+                result.ContentTypes.Add("application/problem+json");
+                result.ContentTypes.Add("application/problem+json");
```
Comment (khellang) [inline src/Microsoft.AspNetCore.Mvc.Core/Internal/ApiBehaviorOptionsSetup.cs]: Was this intentional?

Comment (rynowak) [reply src/Microsoft.AspNetCore.Mvc.Core/Internal/ApiBehaviorOptionsSetup.cs]: If this is still here, it's wrong. We're added a class definition for `problem` but we're not actually using it. Thanks for catching this.

Comment (pranavkm) [reply src/Microsoft.AspNetCore.Mvc.Core/Internal/ApiBehaviorOptionsSetup.cs]: I spoke to @Eilon about this. I missed changing this (I thought I did) and he didn't think it met the ask-mode bar to fix at this point.

**Pull request closed without merging**